File: app/src/main/java/zmuzik/ubike/di/MainScreenModule.java
package zmuzik.ubike.di;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import zmuzik.ubike.MainScreenPresenter;
import zmuzik.ubike.MainScreenView;

@Module
public class MainScreenModule {

    private final Activity mActivity;

    public MainScreenModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mActivity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }

    @Provides
    @ActivityScope
    MainScreenPresenter providePresenter() {
        return new MainScreenPresenter();
    }

    @Provides
    @ActivityScope
    MainScreenView provideView() {
        return (MainScreenView) mActivity;
    }
}

File: app/src/main/java/zmuzik/ubike/di/MainScreenModule.kt
package zmuzik.ubike.di

import android.app.Activity
import android.content.Context
import dagger.Module
import dagger.Provides
import zmuzik.ubike.MainScreenPresenter

@Module
class MainScreenModule(private val mActivity: Activity) {

    @Provides
    internal fun provideContext(): Context {
        return mActivity
    }

    @Provides
    internal fun provideActivity(): Activity {
        return mActivity
    }

    @Provides
    @ActivityScope
    internal fun providePresenter(): MainScreenPresenter {
        return MainScreenPresenter()
    }
}
